### PR TITLE
PAS0044: Onboarding Parachains - Slots

### LINKED ISSUE #11: RFC to define slot time

### PR DESCRIPTION
**Pull request by al3mart**
PAS Proposal pointing the details of onboarding a parachain to Paseo. 

Closes #11

### CODE REVIEW

File: pas/PAS-44-Onboard-paras-slots.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+## Summary
+Means to onboard a parachain to Paseo via **leases and slot assingment**.
+This PAS is to be deprecated once parachain scheduling via coretime is deployed in Paseo.
+
+## Abstract
+In this document users can find the steps to follow to register their parachain in Paseo.
+- Maintainers of a live parachain in Polkadot or Kusama will be granted a slot in Paseo with no expiration date.
+- Maintainers of parachains in development, or those that have not yet secured a slot in Kusama or Polkadot, will be assigned a lease for a shorter period of time.
```
Comment (hbulgarini) [inline pas/PAS-44-Onboard-paras-slots.md:16]: I think we should make the decision about what is a "shorter period of time". 1 week / 2 weeks ? 
We need to take into account the amount of available validators.

CC: @itsbirdo @educlerici-zondax

**Review by hbulgarini — CHANGES_REQUESTED** [review]

File: pas/PAS-44-Onboard-paras-slots.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+PAS: 0044
+title: Onboarding Parachains - Slots
+status: Proposed
+author: Alejandro (@al3mart)
+created: 27-02-2024
+---
+
+## Summary
+Means to onboard a parachain to Paseo via **leases and slot assingment**.
```
Comment (ltfschoen) [inline pas/PAS-44-Onboard-paras-slots.md:10]:
```suggestion
Means to onboard a parachain to Paseo via **leases and slot assignment**.
```

File: pas/PAS-44-Onboard-paras-slots.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- Maintainers of parachains in development, or those that have not yet secured a slot in Kusama or Polkadot, will be assigned a lease for a shorter period of time.
+
+## Motivation
+At the moment of writing there is no way a user without access to root origin can get their parachain's block finalised by Paseo.
+
+## Specification
+### Overview
+
+Users can do a regular registration via dispatching the extrinsics `registrar.reserve` to reserve a `ParaId` and `registrar.register` to submit their code and state. 
+To obtain a slot for your parachain, aka upgrading your parathread to a parachain, users can log an issue in [Paseo's support repository](https://github.com/paseo-network/support) for what they can use the custom issue tempalte in https://github.com/paseo-network/support/issues/new/choose.
```
Comment (ltfschoen) [inline pas/PAS-44-Onboard-paras-slots.md:25]:
```suggestion
To obtain a slot for your parachain, aka upgrading your parathread to a parachain, users can log an issue in [Paseo's support repository](https://github.com/paseo-network/support) for what they can use the custom issue template in https://github.com/paseo-network/support/issues/new/choose.
```

File: pas/PAS-44-Onboard-paras-slots.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+This might be handy in case the root action is needed in the relay chain, for instance.
+Users can always open an issue in case their desired `ParaId` is < `4000`.
+Resources will be assigned as long as the network is not at capacity. This depends on the number of validators supporting the network.
+For this test environment we work with the assumtion that there should always be at least 2 validators per parachain running in the network.
```
Comment (ltfschoen) [inline pas/PAS-44-Onboard-paras-slots.md:29]:
```suggestion
For this test environment we work with the assumption that there should always be at least 2 validators per parachain running in the network.
```

File: pas/PAS-44-Onboard-paras-slots.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+created: 27-02-2024
+---
+
+## Summary
+Means to onboard a parachain to Paseo via **leases and slot assignment**.
+This PAS is to be deprecated once parachain scheduling via coretime is deployed in Paseo.
+
+## Abstract
+In this document users can find the steps to follow to register their parachain in Paseo.
+- Maintainers of a live parachain in Polkadot or Kusama will be granted a slot in Paseo with no expiration date.
```
Comment (wilwade) [inline pas/PAS-44-Onboard-paras-slots.md:15]: Do we need to have a way to remove dead slots? (perhaps that would be a separate PAS)

An example line might be:

Parachains that have not produced a block in the last 14 days may lose their slot and need to reapply.

**Review by wilwade — COMMENTED** [review]
Two notes around length of slots, but otherwise appears straightforward and easy to follow.

File: pas/PAS-44-Onboard-paras-slots.md (hunk 5 above)
Comment (hbulgarini) [reply pas/PAS-44-Onboard-paras-slots.md:15]: Yes, definitely we should add that rule as part of the Operation and Maintenance process. I would add that task in this existing PAS:

[Paseo Testnet Process - Operation and Maintenance](https://github.com/paseo-network/paseo-action-submission/blob/main/pas/core-support-model.md)

Comment (hbulgarini) [reply pas/PAS-44-Onboard-paras-slots.md:15]: https://github.com/paseo-network/paseo-action-submission/issues/47

File: pas/PAS-44-Onboard-paras-slots.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+Resources will be assigned as long as the network is not at capacity. This depends on the number of validators supporting the network.
+For this test environment we work with the assumption that there should always be at least 2 validators per parachain running in the network.
+
+_If you are willing to support the network, please see [PAS#1](https://github.com/paseo-network/paseo-action-submission/blob/main/pas/PAS_ID1_onboard_infrastructure_providers.md).
+
+### Rationale
+
+The onboarding process is bounded by the resources available in the network. Meaning that we will prioritize parachains that are already live and need a testbed for their users, while trying to
+maintain a healthy ratio of teams that just need sporadic resources from Paseo.
+Teams having secured a slot in either Kusama or Polkadot will receive leases for the duration of **1 year**.
```
[the remark this answers is not in the document]
Comment (hbulgarini) [reply pas/PAS-44-Onboard-paras-slots.md:37]: > I might suggest that the expiration matches the slot expiration on Polkadot/Kusama or perhaps that slot expiration + 30 days or something.

100% agree. 

Also the "no expiration" rule should be removed.

File: pas/PAS-44-Onboard-paras-slots.md (hunk 5 above)
Comment (hbulgarini) [reply pas/PAS-44-Onboard-paras-slots.md:15]: @al3mart Let's remove this and apply what Will recommended here: https://github.com/paseo-network/paseo-action-submission/pull/44#discussion_r1511142019

**Comment by hbulgarini** [comment]
As per internal discussion in the Paseo core team we are determining that parachains on Paseo that are testing the protocol will have the lease for 2 weeks.  Then parachains already active on Polkadot/Kusama will have a longer period of 1 year. 

Of course all this will be revisited soon with coretime.

**Comment by hbulgarini** [comment]
@al3mart please proceed with the changes requested so we can merge this one.

**Comment by al3mart** [comment]
Feedback has been addressed.

As per the conversations we had and until on-demand scheduling arrives to Paseo; 1 year worth of LPs will be assigned to teams who have a live network running in either Kusama or Polkadot, and at most two weeks worth of LPs to teams who don't have their parachains running live yet.

File: pas/PAS-44-Onboard-paras-slots.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+- Maintainers of parachains in development, or those that have not yet secured a slot in Kusama or Polkadot, will be assigned a lease for a shorter period of time: 2 weeks long.
+
+## Motivation
+At the moment of writing there is no way a user without access to root origin can get their parachain's block finalised by Paseo.
+
+## Specification
+### Overview
+
+Users can do a regular registration via dispatching the extrinsics `registrar.reserve` to reserve a `ParaId` and `registrar.register` to submit their code and state. 
+To obtain a slot for your parachain, aka upgrading your parathread to a parachain, users can log an issue in [Paseo's support repository](https://github.com/paseo-network/support) for what they can use the custom issue template in https://github.com/paseo-network/support/issues/new/choose.
```
Comment (wilwade) [inline pas/PAS-44-Onboard-paras-slots.md:31]: Is the `what` supposed to be `that`?

```suggestion
To obtain a slot for your parachain, aka upgrading your parathread to a parachain, users can log an issue in [Paseo's support repository](https://github.com/paseo-network/support) for that, they can use the custom issue template in https://github.com/paseo-network/support/issues/new/choose.
```

**Review by wilwade — APPROVED** [review]
Looks good to me. One optional text cleanup.

**Comment by itsbirdo** [comment]
@hbulgarini I still believe we should offer 30d not 14d for new slots. Less overhead for the teams and 14d extra doesn't hurt us really imo

**Comment by al3mart** [comment]
Could we get this approved in parallel ?
@itsbirdo @hbulgarini @carlosala

**Comment by al3mart** [comment]
> @hbulgarini I still believe we should offer 30d not 14d for new slots. Less overhead for the teams and 14d extra doesn't hurt us really imo

I don't see why we shouldn't allow for longer runs while the network has capacity for it.

File: pas/PAS-44-Onboard-paras-slots.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+## Motivation
+At the moment of writing there is no way a user without access to root origin can get their parachain's block finalised by Paseo.
+
+## Specification
+### Overview
+
+Users can do a regular registration via dispatching the extrinsics `registrar.reserve` to reserve a `ParaId` and `registrar.register` to submit their code and state. 
+To obtain a slot for your parachain, aka upgrading your parathread to a parachain, users can log an issue in [Paseo's support repository](https://github.com/paseo-network/support) for that, they can use the custom issue template in https://github.com/paseo-network/support/issues/new/choose.
+This might be handy in case the root action is needed in the relay chain, for instance.
+Users can always open an issue in case their desired `ParaId` is < `4000`.
```
Comment (hbulgarini) [inline pas/PAS-44-Onboard-paras-slots.md:33]:
```suggestion
Users can always open an issue in case their desired `ParaId` being `2000`> ID < `4000`.
```

**Review by hbulgarini — CHANGES_REQUESTED** [review]

**Review by hbulgarini — APPROVED** [review]

File: pas/PAS-44-Onboard-paras-slots.md (hunk 8 above)
Comment (al3mart) [reply pas/PAS-44-Onboard-paras-slots.md:33]: Users can only reserve `ParaId`s from `4000`. The idea being that parachains running in Rococo can mgirate to Paseo avoiding overlaps

**Pull request merged**